plan_delegator/src/plan_delegator.cpp

```diff
@@ -189,11 +190,22 @@ namespace plan_delegator
                                                                 plan_req.response.trajectory_plan.trajectory_points.begin(),
                                                                 plan_req.response.trajectory_plan.trajectory_points.end());
                 
+                // Assign the trajectory plan's initial longitudinal velocity based on the first tactical plugin's response
+                if(first_trajectory_plan == true)
+                {
+                    latest_trajectory_plan.initial_longitudinal_velocity = plan_req.response.trajectory_plan.initial_longitudinal_velocity;
+                    first_trajectory_plan = false;
+                }
+
                 if(isTrajectoryLongEnough(latest_trajectory_plan))
                 {
                     ROS_INFO_STREAM("Plan Trajectory completed for " << latest_maneuver_plan_.maneuver_plan_id);
                     break;
                 }
+
+                // Update the maneuver plan index based on the last maneuver index converted to a trajectory
+                // This is required since inlanecruising_plugin can plan a trajectory over contiguous LANE_FOLLOWING maneuvers
+                current_maneuver_index = plan_req.response.related_maneuvers.back() + 1; 
```

Comment:
Since we've just validated "isTrajectoryLongEnough", are we safe to assume that `plan_req.response.related_maneuvers` is non-empty? Might not hurt to validate that it's set correctly, as calling `.back()` on an empty list can cause undefined behavior.

Reply:
I agree that it is a good idea to add this check. Fixed.